yeonjae/baekjoon/정렬/1181-단어정렬.js

```diff
@@ -0,0 +1,25 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  input.shift();
+
+  input.sort((a, b) => {
+    return a.length - b.length || a.localeCompare(b);
+  });
+
+  const sort = new Set(input);
+
+  console.log(Array.from(sort).join("\n"));
```

Comment:
```suggestion
  const words = input.slice(1);

  words.sort((a, b) => {
    return a.length - b.length || a.localeCompare(b);
  });

  const wordsSet = new Set(words);

  console.log([...wordsSet].join("\n"));
```
- 풀이 잘봤습니다 ~ 🙇‍♂️
- sort라는 함수명으로 변수를 짓는 것은 지양해주세요!! 또한 set으로 활용하기 떄문에 의도와도 맞지 않는 변수명입니다..🥲
- localCompare를 이용해서 sort를 깔끔하게 정의한건 너무 좋았습니다.
- shift, sort를 통해서 기존 input 배열 자체를 변경하다보니 변수명을 깔끔하게 가져가기 까다로워서, shift는 제거하고 대신 slice를 이용해서 `words`라는 변수를 만들고 이를 활용했습니다.
- 그리고 Array.from도 좋지만 set을 배열로 변환시킬 때는 `전개 연산자`가 더 깔끔한 것 같습니다 ~!

Reply:
넵,변수명 좀 더 고려해서 지어보겠습니다🥲 코멘트 감사합니다!